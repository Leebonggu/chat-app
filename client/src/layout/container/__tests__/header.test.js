import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Header from '../header';

jest.mock('../../../common/link', () => 'Link');
const mockStore = configureStore([thunk]);

test('Header Container should render given props', () => {
  const initialState = {};
  const store = mockStore(initialState);
  const component = renderer.create(
    <Provider store={store}>
      <Header />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
