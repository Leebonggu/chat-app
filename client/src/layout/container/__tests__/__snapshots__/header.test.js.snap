// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Container should render given props 1`] = `
Array [
  <div
    className="nav-header-placeholder"
  />,
  <header
    className="nav-header-container sticky"
  >
    <div
      className="nav-header"
    >
      <div
        className="nav-header-left"
      >
        <a
          aria-label="steep-label"
          href="/"
          title="steep-label"
        >
          <img
            alt="afhi-logo"
            height="46px"
            src="afhiLogo.png"
          />
        </a>
      </div>
      <div
        className="nav-header-right"
      >
        <div
          className="nav-interactions-holder"
        >
          <div
            className="nav-search-bar-wrap"
          >
            <div
              className="nav-interactions-search-icon"
            >
              <svg
                className="search-icon"
                height="24"
                onClick={[Function]}
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  fill="#5a6872"
                />
              </svg>
            </div>
            <div
              className="nav-search-input-wrapper"
            >
              <div
                className="form-item-wrap"
              >
                <div
                  aria-controls="react-autowhatever-navbar-search-input"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-owns="react-autowhatever-navbar-search-input"
                  role="combobox"
                >
                  <div>
                    <label
                      className="sr-only"
                      htmlFor="navbar-search-input"
                    >
                      Search Recreation.gov
                    </label>
                    <input
                      aria-autocomplete="list"
                      aria-controls="react-autowhatever-navbar-search-input"
                      aria-label="Enter text to search and press arrow keys"
                      autoComplete="off"
                      id="navbar-search-input"
                      onChange={[Function]}
                      placeholder="What are you looking for?"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    className="search-suggestions-container"
                    id="react-autowhatever-navbar-search-input"
                    role="list"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="nav-links-wrap"
          >
            <div
              className="nav-help"
            >
              <Link
                aria-label="help"
                href="https://gogole.com"
                rel="noopener noreferrer"
                style={
                  Object {
                    "marginRight": "2.4rem",
                  }
                }
                target="_blank"
              >
                <span
                  className=""
                  style={
                    Object {
                      "color": "black",
                      "fontSize": "1.6rem",
                      "fontWeight": "600",
                    }
                  }
                >
                  Help
                </span>
              </Link>
            </div>
            <div
              className="nav-account-panel"
            >
              <div
                className="nav-login-route"
              >
                <Link
                  aria-label="Sign up"
                  style={
                    Object {
                      "marginRight": "2.4rem",
                    }
                  }
                  to="/sign-up"
                >
                  <span
                    className=""
                    style={
                      Object {
                        "color": "black",
                        "fontSize": "1.6rem",
                        "fontWeight": "600",
                      }
                    }
                  >
                    Sign up
                  </span>
                </Link>
              </div>
              <div
                className="nav-login-route"
              >
                <Link
                  aria-label="Log In"
                  style={
                    Object {
                      "marginRight": "0",
                    }
                  }
                  to="/log-in"
                >
                  <span
                    className=""
                    style={
                      Object {
                        "color": "black",
                        "fontSize": "1.6rem",
                        "fontWeight": "600",
                      }
                    }
                  >
                    Log In
                  </span>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>,
]
`;
